File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    key_hash = {}
    for key in keymap:
        for idx, k in enumerate(key):
            idx += 1
            if k not in key_hash:
                key_hash[k] = idx
            else:
                key_hash[k] = min(idx, key_hash[k])
    
    for target in targets:
        count = 0
        for t in target:
            if t in key_hash:
                count += key_hash[t]
            else:
                count = -1
                break
                
        answer.append(count)
    
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.swift
import Foundation

func solution(_ keymap:[String], _ targets:[String]) -> [Int] {
    let maximumTyping = 999
    var minTypingDict = [Character: Int]()
    var result = [Int]()
    
    for key in keymap {
        for (idx, c) in key.enumerated() {
            minTypingDict[c] = min(idx + 1, minTypingDict[c, default: maximumTyping])
        }
    }

    for target in targets {
        var sum = 0
        for t in target {
            guard let count = minTypingDict[t] else {
                sum = -1
                break
            }
            sum += count
        }
        result.append(sum)
    }
    
    return result
}